File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution {
    public long solution(long n) {
        long answer = 0;
        double a = Math.sqrt(n);
        if(a % 1 != 0.0){
            answer = -1;
        }else{
            long b = (long)a;
            b++;
            answer = b * b;
        }
        return answer;
    }
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.swift
import Foundation

func solution(_ n:Int64) -> Int64 {
    var result:Int64 = 0
    var a:Double = sqrt(Double(n))
    if a.truncatingRemainder(dividingBy: 1.0) != 0.0{
        result = -1
    }else {
        let num = Int(a) + 1
        result = Int64(num * num)
    }
    return result
}